frontend/src/interfaces/types.ts

```diff
@@ -1,36 +1,36 @@
-import { NavigateFunction } from 'react-router-dom'
+import { NavigateFunction } from 'react-router-dom';
 
 export interface RegisterData {
-  email: string
-  password: string
-  firstName: string
-  lastName: string
+  email: string;
+  password: string;
+  firstName: string;
+  lastName: string;
 }
 
 export interface LoginData {
-  email: string
-  password: string
+  email: string;
+  password: string;
 }
 export interface ContextTypes {
-  user: UserType | null | undefined | {}
+  user: UserType | null | undefined;
   login: (
     data: LoginData,
     onSuccess: (nav: NavigateFunction) => void,
     onError: (error: string) => void
-  ) => void
+  ) => void;
```

Comment:
Investigate if there is a better way to implement function types.

Reply:
This seems to be the default way according to docs
[https://www.typescriptlang.org/docs/handbook/2/functions.html#function-type-expressions](url)

Reply:
Ok